### DISCUSSION TITLE
Joomla 4 Support.

### DISCUSSION THREAD

**Discussion by philecom** [joomengine/Joomla-Component-Builder#829] [Q&A] [answered]
Hello guys,
I will like to know which version of JCB supports Joomla 4 Components development.
I compiled a few with JCB 2.12.10 on my Joomla 3x site but couldn't install the compiled component on a Joomla 4x site.

Thank you.

**Comment by TLWebdesign** [comment]
Hi, At the moment Joomla 4 support is still in development. If you like to stay up to date, we've got a JUG Meeting planned for 2hd of December where more details about JCB & J4 progress will be shared. If you can make it you should definitely join it. 

More info on the meeting: https://github.com/vdm-io/Joomla-Component-Builder/discussions/826

Kind regards, Tom

**Comment by jfquestiaux** [comment]
Hi.
I missed the JUG meeting, so I'd like to know what is the current J4 situation. How long and how much would you think it will take to get a J4 version of JCB?

**Comment by kreativejuices** [comment]
Hi All,

Is there any update of the J4 compatibility ?

Thanks

**Comment by Pantaleaogc** [comment]
Hello everybody..!
Can we have a mere expectation of the JCB upgrade to Joomla4?
We greatly appreciate your attention

**Reply by jeejeestudio** [reply]
I've read somewhere, here or on the website, that it was scheduled to be released at the end of this year.

**Comment by Llewellynvdm** [comment]
There is two helpful videos on this...

The _fun_ update in May https://youtu.be/ZB-HJBSQb6Y

And this interview done today as reflection from our last JUG meeting https://youtu.be/xu8GIiqOlv0

To stay on-top of the latest news, join us on Telegram: https://t.me/Joomla_component_builder

**Comment by kreativejuices** [comment]
ok thanks for the update, I've just made a contribution on open collective. It's not much but i guess anything helps :) looking forward to the J4 version! Please keep us updated.

**Comment by kreativejuices** [comment]
Hi All, any update on the J4 progress? Estimated dates? Also, will it be possible to upgrade an existing site in j3 with jcb to j4 with jcb once the new j4 is released?

**Comment by TLWebdesign** [comment]
Hi, the new estimated date is around januari 2023. The other question about upgrading i don't have an answer for.

**Reply by viocassel** [reply]
Great! We are really looking forward to this release!

**Reply by iam-crg** [reply]
> Hi, the new estimated date is around januari 2023. The other question about upgrading i don't have an answer for.

Any further updates?

**Comment by jeejeestudio** [comment]
Super mooi! - Super great!
Kan niet wachten :) - Can't wait :)

I'm writing a new project in PHP, but I'd like to put that into a component for jml4. 

Thanks for all you do!

**Comment by kreativejuices** [comment]
Hi All, anyone heard any update on this?

**Comment by vfxhost** [comment]
Hello,
Since the support has ended for Joomla 3, do we have a new estimated date for Joomla 4 support?
Thanks for all your great work so far!

**Reply by kreativejuices** [reply]
I've abandoned it for now. I uninstalled JCB from all my sites and manually updated my components i made to J4 compatibilty and then upgraded the sites to J4. I will wait to see what happens in the future with the development of JCB and make a decision then if I ever come back to it. I've been using Joomla Component Creator instead for now. It's not as powerful but it lays down the base framework needed for J4 components.

**Reply by vfxhost** [reply]
JCB is far more powerful than any other competitor out there. I don't want to abandon it yet. I am sorry that I can't financially support the project and I understand how difficult it is as a spare-time project.  I am sure the creator is doing his best to adapt it to Joomla 4. 
I don't know if temporarily removing some JCB functions will make it easier to compile components for Joomla 4

**Reply by kreativejuices** [reply]
Just taking a guess i would assume it's not that easy as the conversion to J4 has been in development since 2021.  I totally agree that its the most powerful well thought out option available for J3. But unfortunately I can no longer host any J3 websites on my servers so I have no choice.

**Reply by Llewellynvdm** [reply]
I am working diligently on this, and we've made significant progress. Our goal is to have a working prototype by September, but there's still a lot to accomplish. For those interested in assisting with testing, we have a beta channel available at https://git.vdm.dev/joomla/Component-Builder/wiki/Beta-Testing.

Soon, we'll be ready. With this advancement, we aim to transform JCB so it won't be rigidly tied to a single version of Joomla. In the future, you'll be able to adapt your component architecture independently, which is one of our primary objectives. Thank you for your continued interest and support!

**Reply by jeejeestudio** [reply]
This link on the Beta testing page leads to the version for Joomla 3, is there a link to the Beta to test for Joomla 4?

Thanks!

**Reply by Llewellynvdm** [reply]
While testing is still on Joomla 3, the changes made and the features added are all steps towards a change in architecture. We are a vast community of component developers with numerous extensions tailored for Joomla 3. So, when JCB transitions to Joomla 4, all our extensions should auto-upgrade (except for custom scripting). This means I'm steering a colossal ship, preparing it to ascend from water and take flight. We're equipping this ship with various flight mechanisms that grant it speed and wings, and within the next few months, we aim to take off.

By testing the beta and adhering to the instructions, you're assisting me in ensuring that the changes I'm implementing won't jeopardize your projects during our transition to Joomla 4. Without this feedback, we risk leaving some of you behind, and then it might be too late. I urge you to help me test JCB now, and point out any oversights I may have made.

**Reply by jeejeestudio** [reply]
Ok, I will check if I can make a start by turning my PHP project into a Joomla component. So, changing it into MVC model.

**Reply by vfxhost** [reply]
Has there been any progress on the Joomla 4 support since the last discussion? Are we closer to having a working prototype or beta version that supports compiling components for Joomla 4?

**Reply by Llewellynvdm** [reply]
Yes, there is huge progress, and I am looking forward to pushing out the next major changes. The last two weeks, I couldn't get to JCB as I had to focus on some paid work. But I hope to get back... and find myself very excited, as I realize the next few weeks are going to be tremendous.

**Reply by dormasta** [reply]
Just a quick question, will the update work with Joomla 5?

**Comment by Llewellynvdm** [comment]
## JCB Beta Release: [Your Feedback Needed](https://git.vdm.dev/joomla/Component-Builder/issues/1025)!

🚀 Exciting news: JCB's beta version is out! We've made strides, but need your help to perfect it, especially in the list, tag, and category fields.

🐞 Key Updates & Challenges:
- Styling improvements, but some areas need more work.
- Encountering bugs in complex views.
- Import/Export functionality in the backend is problematic.
- Huge strides in adapting to Joomla 4 and 5, but some functions like batch operations aren't ready.
- Good news: Versioning works well, and API integration is in the pipeline.

🙏 Apologies for the delay. Our focus has shifted to ensure JCB is future-proof for Joomla 6, 7, and beyond.

🔍 Testing & Participation:
- Test the beta version and report issues/suggestions on GitHub (tag them as beta-related).
- Join our upcoming code sprint to tackle bugs and plan enhancements (date TBA).

💡 Additional Info:
- Dashboard feature must be active for optimal performance.
- We're resolving naming convention issues with component names.

👥 Your Role is Key:
- Test thoroughly and report each issue separately.
- Use the format [BETA] Title of the Issue for beta-related problems.

Together, let's shape JCB for Joomla 4 and 5. Thanks for your support and collaboration!

Beta JCB v3.1.31 is now out and can be tested: https://git.vdm.dev/joomla-beta/pkg-component-builder/archive/v3.1.31.zip

🤝 [Let's advance together!](https://git.vdm.dev/joomla/Component-Builder/issues/1025)

Took me 30 minutes to update [this project](https://git.vdm.dev/joomla/eHealth-Portal/src/branch/5.0-dev) (a back-end only project) from **Joomla 3** to **Joomla 5**

**Reply by vfxhost** [reply]
It is great! I'm working on updating a component. So far so good. I note some possible bugs and will post them when I am 100% sure.
Thank you so much!

**Comment by Llewellynvdm** [comment, marked as answer]
JCB [v4.0.0-alpha1](https://git.vdm.dev/joomla/pkg-component-builder/releases/tag/v4.0.0-alpha1) and JCB [v5.0.0-alpha1](https://git.vdm.dev/joomla/pkg-component-builder/releases/tag/v5.0.0-alpha1) is out!

Let the testing begin... it will take a few more alpha's before we see a beta version for bot 4&5. But it is a start, the best is yet to come!
